src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -7,61 +7,265 @@ import {
+  },
+  {
+    lovelaceCardConfig: { type: "map", default_zoom: 6 },
+    includeDomains: ["device_tracker"],
+    name: "Map",
+    numberOfEntities: 1,
+  },
+  {
+    lovelaceCardConfig: {
+      type: "markdown",
+      content:
+        "The **Markdown** card allows you to write any text. You can style it **bold**, *italicized*, ~strikethrough~ etc. You can do images, links, and more. For more information see the [Markdown Cheatsheet](https://commonmark.org/help).",
+    },
+    name: "Markdown",
+    noEntity: true,
+  },
+  {
+    lovelaceCardConfig: { type: "media-control" },
+    includeDomains: ["media_player"],
+    name: "Media Control",
+  },
+  {
+    lovelaceCardConfig: {
+      type: "picture",
+      image:
+        "https://www.home-assistant.io/images/merchandise/shirt-frontpage.png",
+    },
+    name: "Picture",
+    noEntity: true,
+  },
+  {
+    lovelaceCardConfig: {
+      type: "picture-entity",
+      image:
+        "https://www.home-assistant.io/images/merchandise/shirt-frontpage.png",
+    },
+    name: "Picture Entity",
+  },
+  {
+    lovelaceCardConfig: {
+      type: "picture-glance",
+      image:
+        "https://www.home-assistant.io/images/merchandise/shirt-frontpage.png",
+    },
+    numberOfEntities: 2,
+    name: "Picture Glance",
+  },
+  {
+    lovelaceCardConfig: {
+      type: "plant-status",
+    },
+    includeDomains: ["plant"],
+    name: "Plant Status",
+    description: "A card for all the lovely botanists out there.",
+  },
+  {
+    lovelaceCardConfig: { type: "sensor", graph: "line" },
+    includeDomains: ["sensor"],
+    name: "Sensor",
+  },
+  {
+    lovelaceCardConfig: { type: "shopping-list" },
+    name: "Shopping List",
+    description:
+      "The Shopping List Card allows you to add, edit, check-off, and clear items from your shopping list.",
+    noPreview: true,
+  },
+  {
+    lovelaceCardConfig: { type: "thermostat" },
+    includeDomains: ["climate"],
+    name: "Thermostat",
+  },
+  {
+    lovelaceCardConfig: { type: "weather-forecast" },
+    includeDomains: ["weather"],
+    name: "Weather Forecast",
+  },
+  {
+    lovelaceCardConfig: { type: "vertical-stack" },
+    name: "Vertical Stack",
+    description:
+      "Vertical stack allows you to group multiple cards so they always sit in the same column.",
+    noPreview: true,
+  },
+];
+
+const manualCardConfigs: CardPickerConfig[] = [
+  {
+    lovelaceCardConfig: { type: "" },
+    name: "Manual",
+    description:
+      "Need to add a custom card or just want to manually write the yaml?",
+    noEntity: true,
+    noPreview: true,
+  },
 ];
 
 @customElement("hui-card-picker")
 export class HuiCardPicker extends LitElement {
   public hass?: HomeAssistant;
-
+  public lovelace?: LovelaceConfig;
+  public entities?: string[];
   public cardPicked?: (cardConf: LovelaceCardConfig) => void;
+  private filteredCardConfigs?: CardPickerConfig[];
+
+  public connectedCallback(): void {
+    super.connectedCallback();
```

Comment:
Do this in `firstUpdated`

Reply:
I will need to make hass a property to do this. First Updated never fires as there are no properties that are monitored

Reply:
`hass` should be a property